modules/trusted-firmware-m/Kconfig

```
@@ -295,6 +295,18 @@ config TFM_PROFILE_TYPE_MINIMAL
 	  and reproduce the desired configuration through kconfig fragments.
 endchoice
 
+config TFM_NFCT_PINS_AS_GPIOS
+	bool "TF-M configure NFCT pins as GPIOs"
+	depends on SOC_NRF5340_CPUAPP
```

```suggestion
	bool
	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_NFCT))
```

Let's leave this as a promptless option so that the user must configure this through DT in the same way for TF-M and non-TF-M builds.

Unless I'm misunderstanding something?

---

Both nrf54h and nrf54l has this option in the mdk, so depends on SOC_NRF5340_CPUAPP is not very future proof.

Note that it's randomly called NRF_CONFIG_NFCT_PINS_AS_GPIOS and CONFIG_NFCT_PINS_AS_GPIOS in these mdks, so there's probably more issues to be handled there.

---

Your depends actually is great. But we cannot make it promptless because this is configured though the UICR node in DTS. And we don't have a UICR node in the non-secure targets.

---

The usage of NRF_CONFIG_NFCT_PINS_AS_GPIOS seems like a hack to me. So hopefully we can revert this and use only CONFIG_NFCT_PINS_AS_GPIOS, right? 
Are you suggesting to enable this here as well @vili-nordic ?